### PR TITLE
Bug 1862896: Fix CamelCase to snake-case conversion

### LINKED ISSUE #886: CamelCase to snake-case conversion is broken
As a result I cannot run `make generate`.

### Reproducer
```shell
# Use a clean environment
docker run -it --rm golang /bin/bash

# Inside the container clone the repo and run make generate
git clone https://github.com/openshift/cluster-monitoring-operator.git
cd cluster-monitoring-operator
wget https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64; mv jq-linux64 /usr/bin/jq; chmod +x /usr/bin/jq
make generate

# It fails with this message
++ echo alertmanager/alertmanager
++ awk '{
  while ( match($0, /(.*)([a-z0-9])([A-Z])(.*)/, cap))
      $0 = cap[1] cap[2] "-" tolower(cap[3]) cap[4];
    print
}'
awk: line 2: syntax error at or near ,
+ fullfile=
make: *** [Makefile:89: build-jsonnet] Error 2
```

### Workaround
Modify the `hack/build-jsonnet.sh`like this:

```diff
-	fullfile=$(echo "${file}" | awk '{
-  while ( match($0, /(.*)([a-z0-9])([A-Z])(.*)/, cap))
-      $0 = cap[1] cap[2] "-" tolower(cap[3]) cap[4];
-    print
-}')
+	fullfile=$(echo "${file}" | sed 's/\(.\)\([A-Z]\)/\1-\2/g' | tr '[:upper:]' '[:lower:]')
```

It seems to be working:

```bash
++ echo thanos-ruler/thanosRulerPrometheusRule
++ sed 's/\(.\)\([A-Z]\)/\1-\2/g'
++ tr '[:upper:]' '[:lower:]'
+ fullfile=thanos-ruler/thanos-ruler-prometheus-rul
```

### PR DESCRIPTION
**Pull request by drpaneas**
Replace 'awk' with 'sed' and change the naming convention of
"c-a" into "ca" for the telemeter-client asset in order to
match with the rest.

Fixes #886

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: hack/build-jsonnet.sh [hunk 1]
```diff
@@ -27,11 +27,7 @@ do
 	path="${prefix}/${dir}"
 	mkdir -p "${path}"
 	# convert file name from camelCase to snake-case
-	fullfile=$(echo "${file}" | awk '{
-  while ( match($0, /(.*)([a-z0-9])([A-Z])(.*)/, cap))
-      $0 = cap[1] cap[2] "-" tolower(cap[3]) cap[4];
-    print
-}')
+	fullfile=$(echo "${file}" | sed 's/\(.\)\([A-Z]\)/\1-\2/g' | tr '[:upper:]' '[:lower:]')
```
Comment (paulfantom) [inline hack/build-jsonnet.sh:30]: Was this tested on OSX?

Comment (drpaneas) [reply hack/build-jsonnet.sh:30]: Yes, this was tested with the latest OSX and also Debian (the golang:latest container).

**Comment by drpaneas** [comment]
Is there anything else needed from my side here?

**Comment by RiRa12621** [comment]
nope
just needs lgtm

**Comment by RiRa12621** [comment]
/assign @s-urbaniak
obey the bots

**Comment by ironcladlou** [comment]
This needs merged to fix generation.

**Comment by s-urbaniak** [comment]
@RiRa12621 @ironcladlou @drpaneas we are using GNU's `awk` variant here available on most linux distros. The reason it fails for the `docker` image is because this one is using `mawk` and OSX uses the BSD variant of `awk` (that landscape is quite messy indeed).

If `sed` is a more common denominator then I am definitely fine with merging this but we should be clear about which variants we expect in the README of this repository given we removed the in-docker generation, cc @paulfantom.

**Comment by s-urbaniak** [comment]
fwiw i at last checked `sed` in the `golang` Docker image uses the GNU variant. That's most likely the reason why it started working for more parties.

**Comment by s-urbaniak** [comment]
putting on hold while @lilic is verifying this works on OSX using BSD's `sed` variant.

**Pull request merged**